internal/webhook/v1/alertmanager_config_secret_webhook.go

```diff
@@ -18,80 +18,145 @@ package v1
-type SecretCustomValidator struct {
-	// TODO(user): Add more fields as needed for validation
+
+type AlertmanagerConfigSecretCustomValidator struct {
+	client client.Client
 }
 
-var _ webhook.CustomValidator = &SecretCustomValidator{}
+var _ webhook.CustomValidator = &AlertmanagerConfigSecretCustomValidator{}
 
 // ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type Secret.
-func (v *SecretCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
 	secret, ok := obj.(*corev1.Secret)
 	if !ok {
 		return nil, fmt.Errorf("expected a Secret object but got %T", obj)
 	}
-	secretlog.Info("Validation for Secret upon creation", "name", secret.GetName())
 
-	// TODO(user): fill in your validation logic upon object creation.
+	log.Info("Validation for Secret upon creation", "name", secret.GetName())
 
-	return nil, nil
+	if err := v.validateTenant(ctx, secret); err != nil {
+		return nil, err
+	}
+	return nil, validateAlertmanagerConfig(ctx, secret)
 }
 
 // ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type Secret.
-func (v *SecretCustomValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
 	secret, ok := newObj.(*corev1.Secret)
 	if !ok {
 		return nil, fmt.Errorf("expected a Secret object for the newObj but got %T", newObj)
 	}
-	secretlog.Info("Validation for Secret upon update", "name", secret.GetName())
 
-	// TODO(user): fill in your validation logic upon object update.
+	log.Info("Validation for Secret upon update", "name", secret.GetName())
 
-	return nil, nil
+	if err := v.validateTenant(ctx, secret); err != nil {
+		return nil, err
+	}
+	return nil, validateAlertmanagerConfig(ctx, secret)
 }
 
 // ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type Secret.
-func (v *SecretCustomValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
-	secret, ok := obj.(*corev1.Secret)
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	// We have nothing to validate on deletion
+	return nil, nil
+}
+
+func (v *AlertmanagerConfigSecretCustomValidator) validateTenant(ctx context.Context, secret *corev1.Secret) error {
+	// Check that the secret has the correct labels.
+	tenant, ok := secret.Labels[tenancy.TenantSelectorLabel]
 	if !ok {
-		return nil, fmt.Errorf("expected a Secret object but got %T", obj)
+		return fmt.Errorf("tenant label is required")
```

Comment:
Please return the full name of the expected label.

```suggestion
		return fmt.Errorf("%s label is required", tenancy.TenantSelectorLabel)
```